core/test/com/google/inject/ProvidedByAndScopeTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2006 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.inject;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+/** @author bn0010100@gmail.com (Neil) */
+public class ProvidedByAndScopeTest {
+  /* static because only support inject to static inner class */
+  static class MixClassProvider implements Provider<MixClass> {
+    public MixClass get() {
+      return new MixClass();
+    }
+  }
+
+  @Singleton
+  @ProvidedBy(MixClassProvider.class)
+  static class MixClass {
+    @Override
+    public boolean equals(Object obj) {
+      return this == obj;
+    }
+  }
+
+  @Test
+  public void testMixClass() throws Exception {
+    final Injector injector =
+        Guice.createInjector(
+            new AbstractModule() {
```

Comment:
You can just call `Injector injector = Guice.createInjector();` without any module for the same value.

Reply:
OK，I will update it today : ) 
Thanks for your advice!